tests/test_admin_digest_router.py

```
@@ -0,0 +1,139 @@
+    def __init__(self) -> None:
+        self.calls: list[tuple[str, str | None]] = []
+
+    async def send_daily_digest(self, payload) -> bool:  # mirrors signature
+        self.calls.append(("send_daily_digest", getattr(payload, "markdown_body", None)))
+        return True
+
+
+class AdminDigestRouterAuthTests(unittest.TestCase):
+    def setUp(self) -> None:
+        # Force a known secret for the duration of these tests
+        os.environ["DIGEST_SECRET"] = "test_digest_secret_at_least_32_chars_long"
```

Ruff (0.15.17)

[error] Possible hardcoded password assigned to: "DIGEST_SECRET"

(S105)